staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -219,6 +219,17 @@ type ObjectMeta struct {
 	ClusterName string `json:"clusterName,omitempty" protobuf:"bytes,15,opt,name=clusterName"`
 }
 
+const (
+	// NamespaceDefault means the object is in the default namespace which is applied when not specified by clients
+	NamespaceDefault string = "default"
+	// NamespaceAll is the default argument to specify on a context when you want to list or filter resources across all namespaces
+	NamespaceAll string = ""
+	// NamespaceNone is the argument for a context when there is no namespace.
+	NamespaceNone string = ""
+	// NamespaceSystem is the system namespace where we place system components.
+	NamespaceSystem string = "kube-system"
```

Comment:
I don't now that I agree.  The convention is that any Kube like API that knows about namespaces knows what `kube-system` is.  How about this, we argue more about where it should end up and then we move it separately (I'd rather not unwind it here).